core/src/main/java/hudson/PluginManager.java

```diff
@@ -1887,6 +1887,20 @@ public HttpResponse doUploadPlugin(StaplerRequest req) throws IOException, Servl
         return FormValidation.ok();
     }
 
+    @Restricted(NoExternalUse.class)
+    @RequirePOST public FormValidation doCheckUpdateSiteUrl(StaplerRequest request, @QueryParameter String value) throws IOException {
+        if (StringUtils.isNotBlank(value)) {
+            try {
+                new URL(value);
```

Comment:
rather than just checking the URL can be parsed should a request be made to see if there is even something serving JSON (html with embedded json 😱 ) at what would be given constructed URL?

Reply:
Oooh, great idea. (Probably needs permission check unless `PluginManager` is entirely protected.)

Reply:
Also needs to be mindful of the changes done to the URL (adding query parameters).

Reply:
@jtnord Update Site also accepts JSONP format. It seems that there's no method to validate the JSONP format😅.

Reply:
a simple `HEAD` for the url with the version parameter appended (`?version=jenkins.version`) and check that you get a valid (non error) HTTP response rather than a timeout would likely catch a number of typos.

Reply:
@jtnord Now the method will send the requests and check if the connection failed (exception thrown).

Reply:
Note that the connection test feature adds the `uctest` query parameter. This serves bandwidth for the infra, because we can send an empty response: https://github.com/jenkinsci/jenkins/blob/237b9ca457c032885ed98e29620f0ed8c0e55ca0/core/src/main/java/hudson/model/UpdateCenter.java#L1208

Reply:
@daniel-beck The parameter is added. Thanks for the suggestion!